Cwbollinger/conversion tools

cwbollinger opened the pull request:
Adds tools for converting hrdf files to urdf and sdf.

cwbollinger committed:
Added tools for generating urdf and sdf files from hrdf

cwbollinger committed:
Added gazebo directives to avoid combining fixed links

Naming conventions in sdf files now align better with hrdf

cwbollinger committed:
Confirmed sdf files working in simulation

Added batch script to generate the 4 gripperless modes

iamtesch commented on sdf/A-2084-01.sdf:
```diff
@@ -0,0 +1,504 @@
+        <geometry>
+          <mesh>
+            <scale>0.001 0.001 0.001</scale>
+            <uri>model://hebi_description/meshes/X5.stl</uri>
```
Did we get these meshes successfully showing up in gazebo when running with these .sdf files?

cwbollinger replied:
Yes, but I should change them to hebi_common, otherwise we have a dependency on hebi_description non-ROS users may want to avoid. Since the meshes never really change this seems like an ok thing to have duplicated?

iamtesch commented on tools/generate_pipeline_batch_v1.bash:
```diff
@@ -0,0 +1,10 @@
+python3 urdf_generator.py ../hrdf/4-DoF_arm_scara.hrdf --urdfdir ../urdf/kits --sdfdir ../sdf --family HEBI --actuators base shoulder elbow wrist1
+python3 urdf_generator.py ../hrdf/4-DoF_arm.hrdf --urdfdir ../urdf/kits --sdfdir ../sdf --family HEBI --actuators base shoulder elbow wrist1
+python3 urdf_generator.py ../hrdf/5-DoF_arm.hrdf --urdfdir ../urdf/kits --sdfdir ../sdf --family HEBI --actuators base shoulder elbow wrist1 wrist2
+python3 urdf_generator.py ../hrdf/6-DoF_arm.hrdf --urdfdir ../urdf/kits --sdfdir ../sdf --family HEBI --actuators base shoulder elbow wrist1 wrist2 wrist3
```
:+1:

iamtesch commented on urdf/end_effector.xacro:
```diff
@@ -2,7 +2,28 @@
 <robot xmlns:xacro="http://www.ros.org/wiki/xacro">
   <!-- TODO: add collision and inertial elements -->
```
Can we remove this TODO, or does this still apply to the parallel gripper subassembly?

iamtesch reviewed (approved):
Looks good!  I added a couple of small comments; feel free to make changes at your discretion.

cwbollinger committed:
Updated sdf folder structure to match standards

model.config file is now generated with sdfs.
gazebo tag for hebi plugin is now added to all models

cwbollinger committed:
Updated hrdf names to match kits, and removed grippers for now
